### DISCUSSION TITLE
Pragma name for @tables streams?

### DISCUSSION THREAD

**Discussion by tbeseda** [architect/architect#1264] [Q&A] [answered]
Currently Arc supports the `@streams` pragma to designate handler functions for DynamoDB streams. It's not documented yet so [I started a doc in arc.codes](https://github.com/architect/arc.codes/pull/450), but that brought up the question:

> Is `@streams` the right name?

Now is a good opportunity to change the pragma and src/streams directory name, if there's a better alternative.

Personally, not having intimate knowledge of the AWS ecosystem when I first read about Arc, I wouldn't know it was related to database event streaming.

Maybe it's "triggers" or "tables-triggers"? If it were to be a "tables-" prefix, what about "@tables-indexes"?

(`stream=true` under a `@tables` entry is how the docs currently demonstrate this feature.)

**Comment by gustmartins** [comment]
I don't think `@streams` is the right name!

First time I read this name I thought it was about Kenesis. 😅 (actualy, if we seach "stream" on aws console search field, first result will be Kenesis)
Only "triggers" would sound too much generic to me.

For me, although I personally dislike prefixes, `@tables-streams` should be perfect for **DynamoDB stream** and changing `@indexes` to `@tables-indexes` would make things more consistent.

This way, by reading the name we could relate "indexes" with "tables" and not think "index for what?"

**Reply by tbeseda** [reply]
I think I agree here. "streams" can be a ton of things, but if it's prefixed with `tables-` it's pretty clear what it relates to 😄

**Comment by macdonst** [comment]
I'm not a expert on databases by any means but I did a quick look at what other vendors call the same functionality:

DB | Term
--- | ---
DynamoDB | Stream
CosmosDB | Change Feed
MongoDB | Activity Feed
Firebase | Event Stream
Cloudant | Stream

So there isn't a consistent name for the functionality but `stream` doesn't seem out of place.

**Reply by tbeseda** [reply]
Some good context!
And I think for just-a-database, it makes sense to use streams. But Arc thinks about all-the-things AWS (maybe other clouds one day), it might be too generic.
I think this is why Mongo's hosted PaaS, Atlas, uses "triggers" for the built in db feed, since other things have "activity".
If the pragma were to include the word "stream" it would need a prefix; at least as a top-level directive. Just about anything can be streamed 😓

**Comment by ryanblock** [comment, marked as answer]
Sounds like `@tables-streams` is the way to go. Thanks y'all!

Re. `@indexes`, I can see possibly aliasing `@tables-indexes`, but we'll probably never fully cut it over – that one is pretty settled, having been there since the beginning of Architect.

**Reply by tbeseda** [reply]
So I'm guessing `@indices` isn't in the cards? kidding! 😉
